File: Contrado.Core/Models/Product.cs
using System;
using System.Collections.Generic;

namespace Contrado.Core.Models
{
    public class Product
    {
        public long ProductId { get; set; }
        public int ProdCatId { get; set; }
        public string ProdName { get; set; }
        public string ProdDescription { get; set; }
        public virtual ProductCategory ProductCategory { get; set; }
        public virtual ICollection<ProductAttribute> ProductAttributes { get; set; }


    }
}

File: ContradoSPA/src/app/Models/Product.ts
export class Product {
  ProductId: number;
  ProdCatId: number;
  ProdName: string;
  ProdDescription: string;
  ProductCategory: ProductCategory;
}

export class ProductDetail {
  ProductId: number;
  ProdCatId: number;
  ProdName: string;
  ProdDescription: string;
  ProductCategory: ProductCategory;
  ProductAttributesDto: ProductAttribute[];
}

export class ProductCategory {
  ProdCatId: number;
  CategoryName: string;
}
export class ProductAttribute {
  ProductId: number;
  AttributeId: number;
  AttributeName: string;
  AttributeValue: string;
}
